src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1000,24 +937,28 @@ private void resetChannelMetadataAfterRecovery(
     // reset the buffer
     this.bufferLock.lock();
     try {
-      LOGGER.warn(
-          "[RESET_PARTITION] Emptying current buffer:{} for Channel:{} due to reset of offsets in"
-              + " kafka",
-          this.streamingBuffer,
-          this.getChannelNameFormatV1());
-      this.streamingBuffer = new StreamingBuffer();
+      // reset buffer by removing already committed records
+      StreamingBuffer resetBuffer = streamingBufferToReset;
+      if (offsetRecoveredFromSnowflake != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
```

Comment:
this is the crux of the change - instead of emptying the buffer and waiting for record.offset == processedOffset + 1, we remove only the records that have been committed, so we can take any record.offset > processedOffset